db_objects/roster_view.class.php

```diff
@@ -460,6 +460,34 @@ public function printCSV($start_date=NULL, $end_date=NULL, $return=FALSE)
 
 		foreach ($to_print as $date => $ddetail) {
```

Comment:
likewise here, it's assuming that there is one service per date, but that's not the case